azsettings/settings.go

```diff
@@ -22,6 +22,8 @@ type AzureSettings struct {
 
 	// This field determines which plugins will receive the settings via plugin context
 	ForwardSettingsPlugins []string
+
+	customClouds []*AzureCloudSettings
```

Comment:
I realized we may need to export this customClouds field after all - I was trying to keep it private to the package to push callers to use the functions, but inside grafana/grafana, we need to write this list back out as a json string when we push settings over to plugins via env variables.  Thoughts?

Reply:
I think this should be fine to remain private for now. In `grafana/grafana` we can still call the `Clouds` or `CustomClouds` functions to retrieve the values.

Reply:
`Clouds()` and `CustomClouds()` return `AzureCloudInfo` struct, which does not have all the fields that are on `AzureCloudSettings` struct.  `grafana/grafana` needs the AzureCloudSettings to get the additional information.  I think we might eventually want to get rid of the `AzureCloudInfo` and just return the full set of fields in one struct and then any caller can use the same exact set of methods for all use cases.  If a caller doesn't need the additional fields, they can ingore them.  Thoughts?  For now, I was just leaning toward exporting `customClouds`.  Thoughts?

Reply:
I'm fine with exporting the field for now, there's nothing that would be considered a secret in there afaict